Makefile.uk: Condition behavior on Kconfig

andreittr opened the pull request:
Previously musl's Makefile would unconditionally:
- reserve a TCB size
- (on AARCH64) unreserve said TCB size
- add musl headers to build include path

, regardless of whether musl was selected for build in Kconfig.
This change set makes these things happen only when musl is selected.

edit: updated w/ new commits

andreittr committed:
Makefile.uk: Condition TCB reserve on Kconfig

Previously musl's Makefile would unconditionally reserve a TCB size,
regardless of whether musl was selected for build in Kconfig.
This change makes this reservation happen only when musl is selected.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

StefanJum reviewed (changes requested):
All good, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

eduardvintila reviewed (commented):
Hey, @andreittr. Good addition! 

While we're at it, I think the `aarch64_no_reserved_tcb_overlap` rule should also include a similar check, so please add one there, as well. Thanks!

andreittr committed:
Condition AARCH64 TCB unreservation on Kconfig

Previously musl would unconditionally call
`aarch64_no_reserved_tcb_overlap` when building under aarch64.
This change makes this call happen only when musl is selected.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

andreittr committed:
Makefile.uk: Condition header inclusion on Kconfig

Previously musl would add its headers to the include path
unconditionally when its Makefile would be included.
This change conditions this inclusion on whether musl is enabled via
Kconfig.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

andreittr commented:
Update: rebased & added new commits

razvand reviewed (commented):
This looks good to me. Tested and it works. @StefanJum, @eduardvintila, please add your review  / approval, and then I'll add the final approval step.

StefanJum reviewed (approved):
All good.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

eduardvintila reviewed (approved):
Thank you!

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

razvand reviewed (approved):
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

andreittr commented:
Rebased on staging.
